R/integrate_profile.R

```diff
@@ -269,44 +289,56 @@ integrate_profile.vpts <- function(x, alt_min = 0, alt_max = Inf,
   stopifnot(is.numeric(alt_min) & is.numeric(alt_max))
   stopifnot(is.na(alpha) || is.numeric(alpha))
 
+  if (alt_max <= alt_min) stop("'alt_min' should be smaller than 'alt_max'")
+
+  # Altitudinal resolution of the bin
   interval <- x$attributes$where$interval
 
-  if (alt_max <= alt_min) stop("'alt_min' should be smaller than 'alt_max'")
+  # Min altitude is the ground level
+  alt_min <- max(alt_min, min(x$attributes$where$height))
 
-  alt_min <- max(alt_min, min(x$height))
+  # Max altitude is the top of the highest bin
   alt_max <- min(alt_max, max(x$height) + interval)
-  if (alt_max - alt_min <= interval) stop(paste("selected altitude range (", alt_min, "-", alt_max, " m) should be wider than the width of a single altitude layer (", interval, " m)", sep = ""))
 
-  index <- which(x$height >= alt_min & x$height < alt_max)
+  # dh is a vector of the height of each bin used for the computation in km
+  # Its value is zeros for bins completly below alt_min or above alt_max, the
+  # interval value for all bin completly above alt_min and below alt_max, and
+  # a ratio of interval for the bins overlaping alt_min and alt_max
+  dh <- pmin(pmin(pmax(x$height-alt_min,0),interval),
+             pmin(pmax(alt_max-x$height,0),interval)) / 1000
+
+  # Vertically Integrated Density in individuals/km^2
+  vid <- colSums(get_quantity(x, "dens") * dh, na.rm = TRUE)
+  # Vertically Integrated Reflectivity in cm^2/km^2
+  vir <- colSums(get_quantity(x, "eta") * dh, na.rm = TRUE)
+
+  # Projection of migratory direction (alpha) on the flight direction (dd)
   if (is.na(alpha)) {
-    cosfactor <- 1 + 0 * get_quantity(x, "dd")[index, ]
+    cosfactor <- 1 + 0 * get_quantity(x, "dd")
   } else {
-    cosfactor <- cos((get_quantity(x, "dd")[index, ] - alpha) * pi / 180)
+    cosfactor <- cos((get_quantity(x, "dd") - alpha) * pi / 180)
   }
 
-  dens_quantity <- colSums(get_quantity(x, "dens")[index, ], na.rm = TRUE)
+  # Migration Traffic Rate in individuals/km/h
+  # multiply speeds (ff) by 3.6 to convert m/s to km/h
+  mtr <- colSums(get_quantity(x, "dens") * cosfactor * get_quantity(x, "ff")
+             * 3.6  * dh, na.rm = TRUE)
+  # Reflectivity Traffic Rate in cm^2/km/h
+  rtr <- colSums(get_quantity(x, "eta") * cosfactor * get_quantity(x, "ff")
+             * 3.6 * dh, na.rm = TRUE)
+
+  # The following quantites are vertically summed by weighting each altitudinal
+  # bin based on their bird reflectivity value (eta) and the height
+  w <- get_quantity(x, "eta") * dh
+  w[is.na(w)] <- 0
+  w <- sweep(w, 2, colSums(w), FUN="/")
```

Comment:
`w` replaces `dens_quantity` by accounting for both the relative quantitfy of bird (`eta`) and the height considered `dh`. 

`sweep` normalized `w` so that the sum of each of its column (i.e. vp) is 1. Its always to simply multiply an sum in l. 336-339.

Reply:
Ok for using `dens `instead of `eta`.
What about `weight_densdh`? I thought it would be good to keep the word "weight", as it is normalized.
Changes pushed in d4ce62d129768eef031e1e581b0443c89883ce01